api/v1beta1/common_types.go

```diff
@@ -53,3 +66,49 @@ type PasswordSelector struct {
 	// Service - Selector to get the watcher service user password from the Secret
 	Service string `json:"service"`
 }
+
+type WatcherImages struct {
+	// +kubebuilder:validation:Required
+	// APIContainerImageURL
+	APIContainerImageURL string `json:"apiContainerImageURL"`
+
+	// +kubebuilder:validation:Required
+	// DecisionEngineContainerImageURL
+	DecisionEngineContainerImageURL string `json:"decisionengineContainerImageURL"`
+
+	// +kubebuilder:validation:Required
+	// DecisionEngineContainerImageURL
+	ApplierContainerImageURL string `json:"applierContainerImageURL"`
+}
+
+func (r *WatcherImages) Default(defaults WatcherDefaults) {
+	if r.APIContainerImageURL == "" {
+		r.APIContainerImageURL = defaults.APIContainerImageURL
+	}
+	if r.DecisionEngineContainerImageURL == "" {
+		r.DecisionEngineContainerImageURL = defaults.DecisionEngineContainerImageURL
+	}
+	if r.ApplierContainerImageURL == "" {
+		r.ApplierContainerImageURL = defaults.ApplierContainerImageURL
+	}
+}
+
+// GetEnvDefault - Get the value associated with key from environment variables, but use baseDefault as a value in the case of an empty string
+func GetEnvDefault(key string, baseDefault string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		return value
+	}
+	return baseDefault
+}
+
+// SetupDefaults - initializes any CRD field defaults based on environment variables (the defaulting mechanism itself is implemented via webhooks)
+
+func SetupDefaults() {
+	// Acquire environmental defaults and initialize Nova defaults with them
+	watcherDefaults := WatcherDefaults{
+		APIContainerImageURL:            GetEnvDefault("WATCHER_API_IMAGE_URL_DEFAULT", WatcherAPIContainerImage),
+		ApplierContainerImageURL:        GetEnvDefault("WATCHER_APPLIER_IMAGE_URL_DEFAULT", WatcherApplierContainerImage),
+		DecisionEngineContainerImageURL: GetEnvDefault("WATCHER_DECISION_ENGINE_IMAGE_URI_DEFAULT", WatcherDecisionEngineContainerImage),
```

Comment:
Should be WATCHER_DECISION_ENGINE_IMAGE_URL_DEFAULT?  note URI vs URL

Reply:
Done!